<!--
    Launch file for testing purposes
    
    After launching this file, it is necessary to publish the
    acquisition command. This can be done as below:
        rostopic pub -1 /RH/cmd/acquire rh_cameras/CamerasSync '{timeStamp: now, data: full}'
        
    SoCS, UoG. Gerardo Aragon-Camarasa. November 2012 
-->

<launch>

    <!-- Include Robot Head parameter server configuration. -->
    <include file="$(find rh_launch)/launch/RH_parameters_$(optenv CLOPEMA_PARTNER error).launch" />
    
    <!-- Configuration file for camera calibration -->
    <param name="/RH/calibration/target" value="opencv" />
    <param name="/RH/calibration/marker_width" value="0.035" />
    <param name="/RH/calibration/marker_size_x" value="7" />
    <param name="/RH/calibration/marker_size_y" value="6" />
    <param name="/RH/calibration/max_error" value="2.0" />
    
    <!-- Which is the arm holding the calibration target (e.g. r1 or r2 - as string) -->
    <param name="/RH/calibration/arm_with_target" value="r2" /> 
    
    <!-- Where to save stuff -->
    <param name="/RH/calibration/outputImageDir" value="$(find rh_calibration)/calibrations/$(optenv CLOPEMA_PARTNER error)/Images/" />
    <param name="/RH/calibration/outputCalibDir" value="$(find rh_calibration)/calibrations/$(optenv CLOPEMA_PARTNER error)/" />
    
    <!-- Saving? 0: Only calib stuff is saved, 1: save processed images in hard disk plus calib stuff, 2: Save everything-->
    <param name="/RH/calibration/save_mode" value="2" type="int"/>
    <!-- Scale ratio in order to reduce the image while finding the inner corners-->
    <param name="/RH/calibration/resize_imgs_factor" value="4.0"/>
    
    <!-- If using alvar -->
<!--    <param name="RH/integration/AlvarMarkerIndexLeft" value="2" type="int"/>-->
<!--    <param name="RH/integration/AlvarMarkerIndexRight" value="2" type="int"/>-->
    
    <!-- Calibration service node -->
    <node pkg="rh_calibration" type="RHcalibration_services" name="RHcalibration_services" output="screen">
    </node>
    
    <!-- For debug -->
    <param name="/RH/calibration/debug" type="bool" value="false" />
    <param name="/RH/calibration/noImagesPTU" value="20" />
    
</launch>

